src/webhook/validation/activegate_test.go

```diff
@@ -149,3 +149,75 @@ func TestMissingActiveGateMemoryLimit(t *testing.T) {
 			})
 	})
 }
+
+func TestExclusiveSyntheticCapability(t *testing.T) {
+	capabilities := []dynatracev1beta1.CapabilityDisplayName{
+		dynatracev1beta1.MetricsIngestCapability.DisplayName,
+		dynatracev1beta1.KubeMonCapability.DisplayName,
+	}
+
+	t.Run(
+		`synthetic-and-another-capability`,
+		func(t *testing.T) {
+			assertDeniedResponse(t,
+				[]string{fmt.Sprintf(errorJoinedSyntheticActiveGateCapability, capabilities[:1])},
+				&dynatracev1beta1.DynaKube{
+					ObjectMeta: defaultDynakubeObjectMeta,
+					Spec: dynatracev1beta1.DynaKubeSpec{
+						APIURL: testApiUrl,
+						ActiveGate: dynatracev1beta1.ActiveGateSpec{
+							Capabilities: append(
+								append(
+									capabilities[:0:0],
+									capabilities[:1]...),
+								dynatracev1beta1.SyntheticCapability.DisplayName),
+						},
+					},
+				})
+		})
+
+	t.Run(
+		`synthetic-surrounded-with-other-capabilities`,
+		func(t *testing.T) {
+			assertDeniedResponse(t,
+				[]string{fmt.Sprintf(errorJoinedSyntheticActiveGateCapability, capabilities)},
+				&dynatracev1beta1.DynaKube{
+					ObjectMeta: defaultDynakubeObjectMeta,
+					Spec: dynatracev1beta1.DynaKubeSpec{
+						APIURL: testApiUrl,
+						ActiveGate: dynatracev1beta1.ActiveGateSpec{
+							Capabilities: append(
+								append(
+									append(
+										capabilities[:0:0],
+										capabilities[:1]...),
+									dynatracev1beta1.SyntheticCapability.DisplayName),
+								capabilities[1:]...),
```

Comment:
`append` accepts variadic arguments, so:
```suggestion
							Capabilities: append(capabilities[:0:0], capabilities[:1]..., dynatracev1beta1.SyntheticCapability.DisplayName, capabilities[1:]...),
```

Reply:
Unfortunately, the **variadic argument** has to be the last on the argument list. The suggested expression offers three variadic and thus is illegal.

Reply:
Oh, ok!
I'd prefer to move appending outside struct instantiation then to reduce level of nesting.

Reply:
Or maybe this?
```
func appendSlices[T any](dst []T, srcs ...[]T) {
	for _, src := range srcs {
		dst = append(dst, src...)
	}
}
```